test/jdk/java/lang/reflect/code/TestConcat.java

```diff
@@ -0,0 +1,240 @@
+import java.lang.reflect.code.op.CoreOps;
+import java.lang.runtime.CodeReflection;
+import java.util.*;
+import java.util.stream.Stream;
+
+/*
+ * @test
+ * @run testng TestConcat
+ */
+
+public class TestConcat {
+
+    static final String TESTSTR = "TESTING STRING";
+
+    static final int TESTINT = 42;
+    @CodeReflection
+    static String f() {
+       int test = 1;
+       //String res = TESTINT + TESTSTR + test;
+       String res = "HI " + TESTSTR + test;
+       return res;
+    }
+
+
+    @CodeReflection
+    public static String byteConcat1(byte b, String s) {
+       return b + s;
+    }
+
+    @CodeReflection
+    public static String byteConcat2(String s, byte b) {
+        return s + b;
+    }
+
+    @CodeReflection
+    public static String shortConcat1(short b, String s) {
+        return b + s;
+    }
+
+    @CodeReflection
+    public static String shortConcat2(String s, short b) {
+        return s + b;
+    }
+
+    @CodeReflection
+    public static String intConcat1(int b, String s) {
+        return b + s;
+    }
+
+    @CodeReflection
+    public static String intConcat2(String s, int b) {
+        return s + b;
+    }
+
+    @CodeReflection
+    public static String longConcat1(long b, String s) {
+        return b + s;
+    }
+
+    @CodeReflection
+    public static String longConcat2(String s, long b) {
+        return s + b;
+    }
+
+    @CodeReflection
+    public static String floatConcat1(float b, String s) {
+        return b + s;
+    }
+
+    @CodeReflection
+    public static String floatConcat2(String s, float b) {
+        return s + b;
+    }
+
+    @CodeReflection
+    public static String doubleConcat1(double b, String s) {
+        return b + s;
+    }
+
+    @CodeReflection
+    public static String doubleConcat2(String s, double b) {
+        return s + b;
+    }
+
+    @CodeReflection
+    public static String booleanConcat1(boolean b, String s) {
+        return b + s;
+    }
+
+    @CodeReflection
+    public static String booleanConcat2(String s, boolean b) {
+        return s + b;
+    }
+
+    @CodeReflection
+    public static String charConcat1(char b, String s) {
+        return b + s;
+    }
+    @CodeReflection
+    public static String charConcat2(String s, char b) {
+        return s + b;
+    }
+    @CodeReflection
+    public static String objectConcat1(Object b, String s) {
+        return b + s;
+    }
+
+    @CodeReflection
+    public static String objectConcat2(String s, Object b) {
+        return s + b;
+    }
+
+    @CodeReflection
+    public static String objectConcat3(TestObject b, String s) {
+        return b + s;
+    }
+
+    @CodeReflection
+    public static String objectConcat4(String s, TestObject b) {
+        return s + b;
+    }
+
+    @CodeReflection
+    public static String stringConcat(String b, String s) {
+        return b + s;
+    }
+
+
+    record Triple(Class<?> first, Class<?> second, String third) {
```

Comment:
Rename to `TestMethodData`?, where the third argument can be moved to the first and renamed to `methodName`? Then you can add functionality on this record to obtain the method from its components and to create the args array.

Reply:
Oddly enough I renamed it locally but I suspect it got caught up in a wave of undos. Thanks for the catch.